File: 0496-next-greater-element-i/0496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        res = [-1] * len(nums1)
        map = {n:i for i,n in enumerate(nums1)}
        
        for i in range(len(nums2)):
            if nums2[i] in map:
                for j in range(i+1, len(nums2)):
                    if nums2[j] > nums2[i]:
                        res[map[nums2[i]]] = nums2[j]
                        break
        return res
        

File: 0496-next-greater-element-i/0496-next-greater-element-i.js
var nextGreaterElement = function(nums1, nums2) {
    let res = nums1.map(x=>-1)
    let map = {}
    for(let i=0;i<nums1.length;i++)
        map[nums1[i]] = i
    
    for(let i=0;i<nums2.length;i++){
        if(map[nums2[i]]!== undefined){
            // console.log(map[nums2[i]])
            
            for(let j=i+1; j<nums2.length; j++){
                if(nums2[j]>nums2[i]){
                    res[map[nums2[i]]] = nums2[j]
                    break
                }
            }
        }
    }
    return res
};